my-site.rb

```diff
@@ -0,0 +1,49 @@
+require "sinatra"
+require "./lib/scrabble.rb"
+
+class MySite < Sinatra::Base
+
+  get "/" do
+    @title = "Homepage"
+    erb :index
+  end
+
+  get "/score" do
+    @title = "Score One Word"
+    erb :score
+  end
+
+  post "/score" do
+    @title = "Word Score"
+    word_array = params[:word].split(" ")
+    word = word_array[0]
+    @valid_word = Scrabble::Scrabble.word?(word)
+    if @valid_word
+      bonus = params[:bonus] == "on"
+      @words = {}
+      @words[word] = {}
+      @words[word][:score] = Scrabble::Scrabble.score(word, bonus)
+      @words[word][:word_breakdown] = Scrabble::Scrabble.get_word_breakdown(word, bonus)
+    end
+    erb :scoredword
+  end
+
+  get "/scoremany" do
+    @title = "Score Many Words"
+    erb :scoremany
+  end
+
+  post "/scoremany" do
+    @title = "Word Score"
+    @valid_word = true
+    word_array = params[:word].split(" ")
+    @words = {}
+    word_array.each do |word|
+      @words[word] = {}
+      @words[word][:score] = Scrabble::Scrabble.score(word)
+      @words[word][:word_breakdown] = Scrabble::Scrabble.get_word_breakdown(word)
```

Comment:
Not sure if this `get_word_breakdown` method was added to Scrabble for this project or if it already exist.

If it was added, kudos on putting that logic in your Scrabble class, it's definitely the best area for generating this kind of data related to the scoring process.

The ideal design I think would have `get_word_breakdown` return an array of the letter scores. The view logic in your template would then display each number from the array with the " + " inserted in between. In this case it's a very minor difference, but the underlying principle is separating that code which relates to the _display_ of data, from that code which relates to the _generation_ of data. We will talk more about this in the coming weeks.